src/OrbitGl/CaptureWindow.cpp

```diff
@@ -113,8 +113,21 @@ void CaptureWindow::PreRender() {
   }
 
   if (time_graph_ != nullptr) {
-    time_graph_->UpdateLayout();
-    UpdateChildrenPosAndSize();
+    const int kMaxLayoutLoops =
+        app_ != nullptr && (app_->IsCapturing() || app_->IsLoadingCapture()) ? 1 : 10;
+
+    int layout_loops = 0;
+
+    // Layout changes of one element may require other elements to be updated as well,
+    // so layouting needs to be done until all elements report that they do not need to
+    // be updated further. As layout requests bubble up, it's enough to check this for
+    // the root element (time graph) of the tree.
+    // During loading or capturing, only a single layouting loop is executed as we're
+    // streaming in data from a seperate thread.
```

Comment:
I don't understand this comment. What is the issue of not updating 10 times? Because performance and that is not fully needed? I would write that instead.

Reply:
Performance mostly, because the layout constantly changes due to addition of new data and sorting of tracks. And we're re-drawing anyways due to the constant addition of data, so it will sort itself out in the next frame.

Reply:
Yes, I suppose that. Please reflect it in a comment.

Reply:
Added a comment, and moved the constant further down